tensorflow-framework/src/test/java/org/tensorflow/framework/losses/CategoricalCrossentropyTest.java

```diff
@@ -0,0 +1,213 @@
+package org.tensorflow.framework.losses;
+
+import org.junit.jupiter.api.Test;
+import org.tensorflow.Operand;
+import org.tensorflow.framework.utils.TestSession;
+import org.tensorflow.ndarray.Shape;
+import org.tensorflow.op.Ops;
+import org.tensorflow.types.TFloat32;
+import org.tensorflow.types.TInt32;
+import org.tensorflow.types.TInt64;
+
+public class CategoricalCrossentropyTest {
+
+  private final TestSession.Mode[] tfModes = {TestSession.Mode.EAGER, TestSession.Mode.GRAPH};
+
+  /** Test of call method, of class CategoricalCrossentropy. */
+  @Test
+  public void testAllCorrectUnweighted() {
+    for (TestSession.Mode tfMode : tfModes)
+      try (TestSession testSession = TestSession.createTestSession(tfMode)) {
+        Ops tf = testSession.getTF();
+
+        long[] trueArray = {
+          1L, 0L, 0L,
+          0L, 1L, 0L,
+          0L, 0L, 1L
+        };
+        float[] predArray = {
+          1.f, 0.f, 0.f,
+          0.f, 1.f, 0.f,
+          0.f, 0.f, 1.F
```

Comment:
The last `f` is in caps. Is that intentional? It's not consistent throughout the file if so.

Reply:
Changed all to `F`